### PR TITLE
LSNBLDR-709 Add calendar component for Lessons page.

### PR DESCRIPTION
**Pull request by nicholaswilson100**

### CODE REVIEW

**Comment by buckett** [comment]
The old PRs are #3660 and #3145

**Comment by ern** [comment]
Should use moment.js from library instead of including here.

**Comment by jonespm** [comment]
Current moment is at
/library/webjars/momentjs/2.11.1/min/moment.min.js

File: lessonbuilder/tool/src/webapp/css/Simplepagetool.css [hunk 1]
```diff
@@ -1473,7 +1473,38 @@ padding:1px;
...
 
+eventAttachmentItem{
+list-style-type: none;
+}
+.fc-event{
+cursor: pointer;
+margin-bottom: 6px; /*to add space between two events*/
+}
+.fc-today-button{
+background-color: #eee !important;
```
Comment (nicholaswilson100) [inline lessonbuilder/tool/src/webapp/css/Simplepagetool.css]: Why are we using all this !important styling? It's not great css practice - can't we do it another way?

There are more uses of it below.

File: lessonbuilder/api/src/java/org/sakaiproject/lessonbuildertool/SimplePageItem.java [hunk 2]
```diff
@@ -46,6 +46,7 @@
 	public static final int QUESTION = 11;
     public static final int BLTI = 12;
     public static final int PEEREVAL = 13;
+    public static final int CALENDAR = 19;
```
Comment (nicholaswilson100) [inline lessonbuilder/api/src/java/org/sakaiproject/lessonbuildertool/SimplePageItem.java]: Move this down a few lines so it's in the right place?

File: calendar/calendar-util/util/src/java/org/sakaiproject/util/CalendarUtil.java [hunk 3]
```diff
@@ -598,5 +605,27 @@ static String getLocalPMString(DateTime now) {
...
+	public Map<String, String> getEventImageMap(Properties configProps){
+		if(eventIconMap.size() == 0){
+			//load keys for locale root
+			Map<String, String> eventKeyMap = new HashMap<String, String>();
+			ResourceBundle bundle = ResourceBundle.getBundle("calendar", Locale.ROOT);
+			for(Enumeration e = bundle.getKeys(); e.hasMoreElements();){
+				String bundleKey = (String)e.nextElement();
+				eventKeyMap.put(bundleKey, bundle.getString(bundleKey));
+			}
+			for(int i=1; i<=20; i++){
```
Comment (nicholaswilson100) [inline calendar/calendar-util/util/src/java/org/sakaiproject/util/CalendarUtil.java]: I'm probably missing something but why 20 iterations?

File: calendar/calendar-util/util/src/java/org/sakaiproject/util/CalendarUtil.java [hunk 4]
```diff
@@ -72,6 +78,7 @@
 	Date dateOctober = null;
 	Date dateNovember = null;
 	Date dateDecember = null;
+	private Map<String, String> eventIconMap = new HashMap<String, String>();
```
Comment (nicholaswilson100) [inline calendar/calendar-util/util/src/java/org/sakaiproject/util/CalendarUtil.java]: Diamond operator?

File: calendar/calendar-tool/tool/src/webapp/WEB-INF/applicationContext.xml [hunk 5]
```diff
@@ -4,11 +4,12 @@
     xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd">
 
 	<bean parent="org.sakaiproject.entitybroker.entityprovider.AbstractEntityProvider"
-		class="org.sakaiproject.calendar.entityproviders.CalendarEventEntityProvider">
+		class="org.sakaiproject.calendar.entityproviders.CalendarEventEntityProvider" init-method="init">
 		<property name="calendarService"
 			ref="org.sakaiproject.calendar.api.CalendarService" />
 		<property name="siteService" ref="org.sakaiproject.site.api.SiteService" />
 		<property name="timeService" ref="org.sakaiproject.time.api.TimeService" />
+		<property name="entityManager"><ref bean="org.sakaiproject.entity.api.EntityManager"/></property>
```
Comment (nicholaswilson100) [inline calendar/calendar-tool/tool/src/webapp/WEB-INF/applicationContext.xml]: Make this the same styling as the other beans ?

File: lessonbuilder/tool/src/webapp/js/show-page.js [hunk 6]
```diff
@@ -470,6 +470,35 @@ $(document).ready(function() {
...
 		});
 
+		$(".edit-calendar").click(function(){
+			oldloc = $(this);
+			closeDropdowns();
+			var row = $(this).closest('li');
+			$("#change-assignment-p").hide();
+			$("#change-quiz-p").hide();
+			$("#change-forum-p").hide();
+			$("#change-resource-p").hide();
```
Comment (nicholaswilson100) [inline lessonbuilder/tool/src/webapp/js/show-page.js]: Is there not a better way to do this with a class or something?

**Comment by buckett** [comment]
@nicholaswilson100 Were you going to address the comments that you made here?

**Comment by nicholaswilson100** [comment]
I think I've addressed all the comments in this PR that need addressing.

**Comment by buckett** [comment]
I think you're still including moment.js in lessonbuilder and it shouldn't be needed any more.

File: calendar/calendar-util/util/src/java/org/sakaiproject/util/CalendarUtil.java (hunk 4 above)
Comment (buckett) [inline calendar/calendar-util/util/src/java/org/sakaiproject/util/CalendarUtil.java]: Still seeing diamond operator here.

Comment (nicholaswilson100) [inline calendar/calendar-util/util/src/java/org/sakaiproject/util/CalendarUtil.java]: I'm happy for this to be merged without this and my other comments being addressed.  They were nice to haves really.

File: lessonbuilder/tool/src/webapp/templates/ShowPage.html [hunk 7]
```diff
@@ -56,13 +56,18 @@
...
  <link rel="stylesheet" href="$context/css/jquery.cluetip.css" type="text/css" />
+ <link type="text/css" href="$context/css/fullcalendar.css" rel="stylesheet" />
  <script type="text/javascript" src="$context/js/peer-eval.js"></script>
  <script type="text/javascript" src="$context/js/hoverIntent.js"></script>
  <script type="text/javascript" src="$context/js/show-page.js"></script>
  <script type="text/javascript" src="$context/js/announcements.js"></script>
  <script type="text/javascript" src="$context/js/gradingAjax.js"></script>
  <script type="text/javascript" src="$context/js/checklistAjax.js"></script>
  <script type="text/javascript" src="$context/js/forum-summary.js"></script>
+ <script type="text/javascript" src="/library/webjars/momentjs/2.11.1/min/moment.min.js"></script>
```
Comment (nicholaswilson100) [inline lessonbuilder/tool/src/webapp/templates/ShowPage.html]: @buckett: do you mean here?

Comment (nicholaswilson100) [inline lessonbuilder/tool/src/webapp/templates/ShowPage.html]: there's a similar line in /syllabus/syllabus-app/src/webapp/syllabus/main.jsp

Comment (nicholaswilson100) [inline lessonbuilder/tool/src/webapp/templates/ShowPage.html]: Oh I see, you mean the file can be deleted.  Yup.

**Comment by lardo-de-arnaud** [comment]
Conflicts here - can U address

**Comment by nicholaswilson100** [comment]
I've rebased this so it has no conflicts

**Comment by nicholaswilson100** [comment]
There aren't any outdated reference/library things in this one.

**Comment by nicholaswilson100** [comment]
@ern Hi Earle,  Not sure if you're the right person to ask about this, but the Travis build for this PR has failed but not because of the code change.  What's the best thing to do?  Can you or I kick it off again or does something need to happen or shall I push an empty file or something? Nick

**Comment by nicholaswilson100** [comment]
It failed because of this test : SearchBeanImplTest

Errors like this:
testGetSearchSitesEmptyValueSiteIdsArray(org.sakaiproject.search.tool.SearchBeanImplTest)  Time elapsed: 0.001 sec  <<< ERROR!
java.lang.NoClassDefFoundError: org/elasticsearch/action/search/SearchResponse
	at org.sakaiproject.search.tool.SearchBeanImplTest.setUp(SearchBeanImplTest.java:32)
Caused by: java.lang.ClassNotFoundException: org.elasticsearch.action.search.SearchResponse
	at org.sakaiproject.search.tool.SearchBeanImplTest.setUp(SearchBeanImplTest.java:32)

**Comment by adrianfish** [comment]
@nicholaswilson100 Bit of badness here. You click on an event and the popup is masked by the tool column.

screen shot 2017-03-14 at 18 33 05

**Comment by adrianfish** [comment]
@nicholaswilson100 I'll push a commit on.

**Comment by nicholaswilson100** [comment]
Amazing thanks!

**Comment by adrianfish** [comment]
I just set it to 400 so it would fit on a mobile screen, too. It seems wide enough.

**Comment by adrianfish** [comment]
Travis build is failing on some elasticsearch tests. When the build succeeds I'll merge this.

**Comment by nicholaswilson100** [comment]
Great thanks :)

**Pull request merged**